### ISSUE TITLE
Add Docker GitHub Actions workflow configuration

### ISSUE THREAD

**Issue by chantzlarge** [go-nostr/nostr#4]
## Prerequisites

- [x] I have searched the existing issues and I'm sure this is not a duplicate.
- [ ] If reporting a bug, I have followed the necessary troubleshooting steps.
- [x] If suggesting a feature, I have explained my reasoning and use case.

### If suggesting a feature:

**Feature description:**

Add a GitHub Actions workflow to automate building and pushing the Docker image to the Docker Hub registry.

**Use case:**

Having a GitHub Actions workflow in place can help streamline the process of building and deploying the application. By automating the build and push of the Docker image to Docker Hub, we can ensure that the latest version of the application is always available for deployment. This also eliminates the need for manual intervention during the build and deployment process, reducing the likelihood of errors and improving overall efficiency.

**Alternative solutions:**

1. Manually building and pushing the Docker image to Docker Hub. This approach can be time-consuming and error-prone, as it requires manual intervention for each update.
2. Using other CI/CD services, such as GitLab CI/CD or Travis CI. These alternatives can achieve similar results but may require additional configuration and integration effort.

**Additional context:**

The GitHub Actions workflow should include the following components:

1. Trigger the workflow on push events to the main branch or when a new release is created.
2. Build the Docker image using the existing `Dockerfile`.
3. Push the built image to Docker Hub with appropriate tags, such as the commit SHA and release version (if applicable).
4. (Optional) Deploy the new Docker image to a staging or production environment upon successful push to Docker Hub.

**Comment by chantzlarge** [comment]
This is complete and functioning as expected up to the client build stage which is failing due to a Angular dependency reference errors. 

[See latest builds]

This Docker configuration using the explicit versions of base image and Angular CLI may be the fix:

https://github.com/docker/awesome-compose/blob/e6b1d2755f2f72a363fc346e52dce10cace846c8/angular/angular/Dockerfile#L8

**Comment by chantzlarge** [comment]
Also noticed the previously mentioned Dockerfile includes a Multi-platform image flag:

https://docs.docker.com/build/building/multi-platform/

**Comment by chantzlarge** [comment]
After some deliberation I discovered the issue was due to the missing subproject package.json dependency. Copying this in addition to the root NPM `package.json` and `package-lock.json` files resolved the issue.

Initially, I copied all of the files before running `npm ci` before later optimizing this by copying the files in addition to the typical `package.json` and `package-lock.json` files.

This is now resolved and functioning as expected. Images are available for pull from ghcr.io.
